test/e2e/load_balancer.go

```diff
@@ -297,3 +296,266 @@ var _ = Describe("ESIPP [Slow]", func() {
+		serviceName := "e2e-tls-lb-test"
+		ns := f.Namespace.Name
+
+		jig := framework.NewServiceTestJig(f.ClientSet, serviceName)
+		nodeIP := framework.PickNodeIP(jig.Client) // for later
+
+		sslSecretName := "ssl-certificate-secret"
+		err := framework.CreateSSLSecretOrFail(f.ClientSet, ns, sslSecretName, []byte(framework.SSLCAData), []byte(framework.SSLCertificateData),
+			[]byte(framework.SSLPrivateData), []byte(framework.SSLPassphrase))
+		framework.ExpectNoError(err)
+		loadBalancerLagTimeout := framework.LoadBalancerLagTimeoutDefault
+		loadBalancerCreateTimeout := framework.LoadBalancerCreateTimeoutDefault
+		if nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet); len(nodes.Items) > framework.LargeClusterMinNodesNumber {
+			loadBalancerCreateTimeout = framework.LoadBalancerCreateTimeoutLarge
+		}
+
+		// TODO(apryde): Test that LoadBalancers can receive static IP addresses
+		// (in a provider agnostic manner?). OCI does not currently
+		// support this.
+		requestedIP := ""
+
+		tcpService := jig.CreateTCPServiceOrFail(ns, func(s *v1.Service) {
+			s.Spec.Type = v1.ServiceTypeLoadBalancer
+			s.Spec.LoadBalancerIP = requestedIP
+			s.Spec.Ports = []v1.ServicePort{v1.ServicePort{Name: "http", Port: 80, TargetPort: intstr.FromInt(80)},
+				v1.ServicePort{Name: "https", Port: 443, TargetPort: intstr.FromInt(80)}}
+			s.ObjectMeta.Annotations = map[string]string{cloudprovider.ServiceAnnotationLoadBalancerSSLPorts: "443",
+				cloudprovider.ServiceAnnotationLoadBalancerTLSSecret:        sslSecretName,
+				cloudprovider.ServiceAnnotationLoadBalancerTLSBackendSecret: sslSecretName}
+
+		})
+
+		svcPort := int(tcpService.Spec.Ports[0].Port)
+
+		By("creating a pod to be part of the TCP service " + serviceName)
+		jig.RunOrFail(ns, nil)
+
+		// TODO(apryde): Test UDP service. OCI does not currently support this.
+
+		By("waiting for the TCP service to have a load balancer")
+		// Wait for the load balancer to be created asynchronously
+		tcpService = jig.WaitForLoadBalancerOrFail(ns, tcpService.Name, loadBalancerCreateTimeout)
+		jig.SanityCheckService(tcpService, v1.ServiceTypeLoadBalancer)
+
+		tcpNodePort := int(tcpService.Spec.Ports[0].NodePort)
+		framework.Logf("TCP node port: %d", tcpNodePort)
+
+		if requestedIP != "" && framework.GetIngressPoint(&tcpService.Status.LoadBalancer.Ingress[0]) != requestedIP {
+			framework.Failf("unexpected TCP Status.LoadBalancer.Ingress (expected %s, got %s)", requestedIP, framework.GetIngressPoint(&tcpService.Status.LoadBalancer.Ingress[0]))
+		}
+		tcpIngressIP := framework.GetIngressPoint(&tcpService.Status.LoadBalancer.Ingress[0])
+		framework.Logf("TCP load balancer: %s", tcpIngressIP)
+
+		By("hitting the TCP service's NodePort")
+		jig.TestReachableHTTP(false, nodeIP, tcpNodePort, framework.KubeProxyLagTimeout)
+
+		By("hitting the TCP service's LoadBalancer")
+		jig.TestReachableHTTP(false, tcpIngressIP, svcPort, loadBalancerLagTimeout)
+
+		// // Change the services' node ports.
```

Comment:
Do we need to be performing all of these checks? Is anything different wrt. TLS or can we rely on the existing tests to check a lot of this?

Reply:
Nope, C&P on my part :)